### PR TITLE
projects/S905: new project

### PR DESCRIPTION
**Pull request by adamg88**

### CODE REVIEW

**Comment by chewitt** [comment]
linux patches should be numbered so the order they are applied in is easily understood

**Comment by chewitt** [comment]
and if possible, patches should be upstreamed to our kernel

**Comment by Raybuntu** [comment]
The missing numbering seems to be an issue for C2, WH and WP2 project specific kernel patches too. Most of the patches exist several times in all those projects.
Upstreaming would be best indeed and would fix the numbering too

File: projects/S905/filesystem/usr/lib/systemd/system/amlogic-wifi.service [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Amlogic WiFi module loader
+
+[Service]
+Type=simple
+ExecStart=/sbin/modprobe wifi_dummy
+
+[Install]
+WantedBy=basic.target
```
Comment (kszaq) [inline projects/S905/filesystem/usr/lib/systemd/system/amlogic-wifi.service]: This service will fail as there is no `wifi_dummy` module in LE kernel.

File: projects/S905/options [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  # additional Firmware to use (dvb-firmware, misc-firmware, wlan-firmware)
+  # Space separated list is supported,
+  # e.g. FIRMWARE="dvb-firmware misc-firmware wlan-firmware"
+    FIRMWARE=""
+
+  # build and install ATV IR remote support (yes / no)
+    ATVCLIENT_SUPPORT="no"
+
+  # "none" for disable LCD support
+    LCD_DRIVER="none"
```
Comment (Kwiboo) [inline projects/S905/options]: `LCD_DRIVER` can be dropped, was moved to an addon in LE8

File: projects/S905/options [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+  # Amlogic IR remote support (yes / no)
+    AMREMOTE_SUPPORT="yes"
+
+  # build and install iSCSI support - iscsistart (yes / no)
+    ISCSI_SUPPORT="no"
+
+  # build with swap support (yes / no)
```
Comment (CvH) [inline projects/S905/options]: swap is now at distribution option
https://github.com/LibreELEC/LibreELEC.tv/commit/4360992b3f565b87742db0eee62c2079933126e8
pls remove the swap related stuff from here (L134-L142)

**Comment by adamg-xda** [comment]
Changes made, thank you

File: projects/S905/options [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+  # additional drivers to install:
+  # for a list of additinoal drivers see packages/linux-drivers
+  # Space separated list is supported,
+  # e.g. ADDITIONAL_DRIVERS="DRIVER1 DRIVER2"
+    ADDITIONAL_DRIVERS="gpu-aml RTL8189FS"
+
+  # additional Firmware to use (dvb-firmware, misc-firmware, wlan-firmware)
+  # Space separated list is supported,
+  # e.g. FIRMWARE="dvb-firmware misc-firmware wlan-firmware"
+    FIRMWARE=""
```
Comment (CvH) [inline projects/S905/options]: `FIRMWARE="misc-firmware wlan-firmware dvb-firmware brcmfmac_sdio-firmware-aml"`
I guess ? (same like at WP2 + OC2 ?)

File: scripts/mkimage [hunk 5]
```diff
@@ -263,6 +263,14 @@ EOF
     mcopy $RELEASE_DIR/target/KERNEL.md5 "::/$KERNEL_NAME.md5"
     mcopy $RELEASE_DIR/target/SYSTEM.md5 ::/SYSTEM.md5
 
+    # compile and copy amlogic autoscript files to part1
```
Comment (Raybuntu) [inline scripts/mkimage]: You build u-boot:host only for S905: https://github.com/LibreELEC/LibreELEC.tv/pull/2315/files#diff-9c7eee3ed0931e5ee6bcba810283ece0R185
So I suggest to check for PROJECT == S905 here too.

File: scripts/mkimage [hunk 6]
```diff
@@ -263,6 +263,14 @@ EOF
...
+    for src in $INSTALL_SRC_DIR/*autoscript.src ; do
+      if [ -e "$src" ] ; then
+        $TOOLCHAIN/bin/mkimage -A $TARGET_KERNEL_ARCH -O linux -T script -C none -d "$src" "$LE_TMP/$(basename $src .src)" > /dev/null
```
Comment (HiassofT) [inline scripts/mkimage]: IMHO it would be cleaner to move the u-boot mkimage calls out of scripts/mkimage, probably to the bootloader package.

**Comment by adamg-xda** [comment]
Ok I have rebased please review new changes, thanks :)

File: projects/S905/options (hunk 4 above)
Comment (adamg-xda) [reply projects/S905/options]: I wasn't sure about the firmwares as they are rather generic repositories, there is no broadcom sdio devices in S905 except WP2 and OC2 as you pointed out and as for dvb-firmware I thought you may be doing something with your media_build changes in relation to that and I wasn't going to tackle dvb internal/external at this point either.

**Comment by Raybuntu** [comment]
Thanks @adamg88 and @kszaq.

**Pull request merged**